instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -51,6 +53,12 @@ public class OpenTelemetryAutoConfiguration {
 
   public OpenTelemetryAutoConfiguration() {}
 
+  @Bean
+  @ConfigurationPropertiesBinding
+  public MapConverter mapConverter() {
```

Comment:
Could this bean be moved to the `OpenTelemetrySdkConfig` class? I don't follow how it works. `mapConverter `bean does not seem to be injected into the OTel code and I don't see how Spring coud reuse this bean. Could you please add a comment?

Reply:
- yes, moved to `OpenTelemetrySdkConfig`
- the converter is used to parse headers, see https://github.com/zeitlinger/opentelemetry-java-instrumentation/blob/4bf15ad6529789eb848f609bf78ae2c96fc04103/instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpSpanExporterAutoConfigurationTest.java#L103-L116

Reply:
I don't follow where the `mapConverter` bean is injected and used. Is the `mapConverter` bean really used (are the tests green without this bean)? The parsing is not done in the `OtlpExporterUtil` class?

Reply:
This test fails when the converter is removed:

https://github.com/zeitlinger/opentelemetry-java-instrumentation/blob/4bf15ad6529789eb848f609bf78ae2c96fc04103/instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpSpanExporterAutoConfigurationTest.java#L109

This is the error message: 

Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'otelOtlpSpanExporter' defined in io.opentelemetry.instrumentation.spring.autoconfigure.exporters.otlp.OtlpSpanExporterAutoConfiguration: Unsatisfied dependency expressed through method 'otelOtlpSpanExporter' parameter 0; nested exception is org.springframework.boot.context.properties.ConfigurationPropertiesBindException: Error creating bean with name 'otel.exporter.otlp-io.opentelemetry.instrumentation.spring.autoconfigure.exporters.otlp.OtlpExporterProperties': Could not bind properties to 'OtlpExporterProperties' : prefix=otel.exporter.otlp, ignoreInvalidFields=false, ignoreUnknownFields=true; nested exception is org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'otel.exporter.otlp.headers' to java.util.Map<java.lang.String, java.lang.String>

Reply:
I got it now. 

Could you please update the code to:
- Move  the `mapConverter` bean to an autoconfiguration only related to OTLP (just realized that it's OTLP specific)
- Only  create the `mapConverter` bean when OTLP is enabled
- Perhaps add a comment to explain that the `mapConverter`  bean is necessary  for the `OtlpExporterProperties` component?

Thank you!

Reply:
I added comments in the different places.

I tried to make the converter local, but that doesn't work:

 for bean 'mapConverter' since there is already [Root bean: class [null]; scope=; abstract=false; lazyInit=null; autowireMode=3; dependencyCheck=0;

Reply:
The `mapConverter` bean is still in the `OpenTelemetryAutoConfiguration` class whose responsibility is to instantiate an `OpenTelemetry` bean.

You could create the `mapConverter` bean with an OTLP-related autoconfiguration.

`OtlpLoggerExporterAutoConfiguration`, `OtlpLoggerExporterAutoConfiguration`, `OtlpSpanExporterAutoConfiguration` are examples of autoconfiguration. 

The autoconfiguration classes have to be declared in the [spring.factories](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/spring.factories) (<Spring Boot 2.7) [org.springframework.boot.autoconfigure.AutoConfiguration.imports](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports) (>=Spring Boot 2.7) files.

Reply:
it's also used to parse `otel.resource.attributes` - which can be seen in the javadoc of the class